scripts/update_dependencies.sh

```diff
@@ -0,0 +1,34 @@
+set -e
+
+if [ "$1" ]
+  then
+    REQ_BRANCH="$1"
+else
+    REQ_BRANCH="develop"
+fi
+
+# make sure bluemira is in the current environment
+BM_INSTALLED=$(pip list | grep bluemira | awk 'NR==1 {print $1}')
+[ "$BM_INSTALLED" = "bluemira" ] || { echo "bluemira not found in the current python environment, exiting" && exit 1; }
+
+# check we are in the bluemira git repo
+[ $(git rev-parse --is-inside-work-tree) = "true" ] ||  { echo "Not in bluemira git repo, exiting" && exit 1; }
+
+CURRENT_REPO=$(git remote get-url origin)
+REPO_GIT_URL="git@github.com:Fusion-Power-Plant-Framework/bluemira.git"
+REPO_HTTPS_URL="https://github.com/Fusion-Power-Plant-Framework/bluemira.git"
```

Comment:
I'm not convinced this is the best way to determine we're in the repo. For most open source projects, `origin` will refer to your _fork_ of the repository, and `upstream` would point to the main repository.

Do we actually need this check? I'd probably opt for getting the base repo directory using `dirname` and use absolute paths everywhere. Then it doesn't matter where in the file system we are, the script always acts the same.

Something like this, maybe?

```bash
readonly BLUEMIRA_ROOT="$(realpath "$(dirname "$0")"/..)"
...
requirements_file="${BLUEMIRA_ROOT}/requirements.txt"
pip install -r "${requirements_file}"
```

Reply:
Fair point! I was just trying to make sure we had the right repo and the right requirements file. Good catch on the origin/upstream thing. Will modify because its probably badly done and overkill protection